vttools/to_wrap/tomo.py

```diff
@@ -0,0 +1,533 @@
+# Redistribution and use in source and binary forms, with or without   #
+# modification, are permitted provided that the following conditions   #
+# are met:                                                             #
+#                                                                      #
+# * Redistributions of source code must retain the above copyright     #
+#   notice, this list of conditions and the following disclaimer.      #
+#                                                                      #
+# * Redistributions in binary form must reproduce the above copyright  #
+#   notice this list of conditions and the following disclaimer in     #
+#   the documentation and/or other materials provided with the         #
+#   distribution.                                                      #
+#                                                                      #
+# * Neither the name of the Brookhaven Science Associates, Brookhaven  #
+#   National Laboratory nor the names of its contributors may be used  #
+#   to endorse or promote products derived from this software without  #
+#   specific prior written permission.                                 #
+#                                                                      #
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS  #
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT    #
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS    #
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE       #
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,           #
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES   #
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR   #
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)   #
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,  #
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+"""
+This is the model for 3D tomography reconstruction
+"""
+
+import sys
+import numpy as np
+import tomopy
+import logging
+
+def _generate_xtomo_object(data=None, white=None, dark=None, theta=None, center=None, recon=None, log_mode='info'):
+    """Generate xtomo_dataset object internally to enable passing only ndarrays
+
+    Parameters
+    ----------
+    data : np.ndarray
+        sample data
+
+    white : np.ndarray
+        white background of the sample
+
+    dark : np.ndarray
+        dark background of the sample
+
+    theta : np.ndarray
+        the angle array of projections
+
+    center : float
+        the center of the projections
+
+    recon : np.ndarray
+        the reconstructed data
+
```

Comment:
In most of our code the ndarray parameter object type is simply ndarray. While we do have an option in the signature map that will handle np.ndarray as a parameter type, the entries in this document should probably be renamed to simply state ndarray for consistency.